code/controllers/subsystem/atoms.dm

```diff
@@ -33,17 +33,18 @@ SUBSYSTEM_DEF(atoms)
 	initialized = INITIALIZATION_INNEW_MAPLOAD
 
 	var/count
-	var/list/mapload_arg = list(TRUE)
+	var/static/list/mapload_arg = list(TRUE)
+	var/atom/A
 	if(atoms)
 		count = atoms.len
-		for(var/I in atoms)
-			var/atom/A = I
+		for(var/I in 1 to count)
+			A = atoms[I]
 			if(!(A.flags_1 & INITIALIZED_1))
-				InitAtom(I, mapload_arg)
+				InitAtom(A, mapload_arg)
 				CHECK_TICK
 	else
 		count = 0
-		for(var/atom/A in world)
+		for(A in world)
```

Comment:
Given the mess that is loop typing, did you check if this affects order.

Reply:
Hummm.  Part of this PR was a combo of another pr that fixes map loading. but that is a good point.  Going to try to see if we can use world.len and do the same as we did with atoms.

Reply:
well fuck me in a hole your right.  God damnit.

Reply:
I hate you, I got to pr this one seperate.

Reply:
Goddamnit. I spent 4 hours profiling it to find out that for(var/atom/A in world) is faster than any other thing I can come up with.  Its so fast in fact its faster for me to have SSnetworks go though all the areas before SSatoms to set their names than to do it during map loading.  Ugh.  Going to refactor some of it and clean it up.

Reply:
And all this is moot point because I discovered this months ago and moved it to SSnetworks anyway.  uggh.  Atleast I found a few bugs.